### ISSUE TITLE
Resynthesizer not showing

### ISSUE THREAD

**Issue by FalcoGer** [bootchk/resynthesizer#95]
I am running gimp 2.10 (`GNU Image Manipulation Program version 2.10.24`) on ubuntu mate 20.04 LTS (`5.4.0-73-generic #82-Ubuntu SMP Wed Apr 14 17:39:42 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux`)

I have tried installing the `gimp-plugins-registry` package but resynthesizer didn't show up. I then downloaded this repository and ran autogen, make and make install. it still doesn't show up in gimp. I then copied the python files and executables to ~/.config/GIMP/2.10/plug-ins and still no dice.

I get Filters/Map/Resynthesize and that does 'something' (seems like it randomly clones or displaces pixels from place to place as i tried to remove some text from an image by selecting the text and running the filter on it), but everything else (heal, enlarge, uncrop, etc) is missing from where [this website](https://www.javatpoint.com/gimp-resynthesizer) says it should be. I tried looking in other menus but found nothing.

**Comment by bootchk** [comment]
I just updated the readme.  Please see it.  It says to use the flatpak of GIMP and the flatpak of the Resynthesizer.

**Comment by FalcoGer** [comment]
I don't want to use any flatpacks. Package repositories are great! Keeps things in one place, updated and you don't have to run all over the internet to get updated software, if you get notified of an update at all in the first place. Besides, flatpacks use unnecessarily space for redundant dependencies.

I mean this is a plugin. Why can't I plug it in like any other plugin?

**Comment by vido89** [comment]
I have same issue Im runing LM 20.04 and GIMP 2.10 and I downloaded and build plugin and place it in `.config/GIMP/2.10/plug-ins` and plugin does not show up in `Filters`. Than I installed flatpak Gimp beta 2.99.6 and Resynthesizer also from flatpak and it does not show up plugin in `Filters`. Any idea ?

**Comment by vido89** [comment]
I just found that PyGTK hasn't been maintained for 9 years, so it's dropped and breaks Resynthesizer for Gimp 2.10.x. So in order to install it you need to install it from flatpak. After installing GIMP run

`flatpak install org.gimp.GIMP.Plugin.Resynthesizer`

but due to the [bug](https://github.com/flatpak/flatpak/issues/4208) you need to copy Resynthesizer from

` /var/lib/flatpak/runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/2-3.36/active/files/plug-ins/`
to
` /home/your-name/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/`

**Comment by bootchk** [comment]
Thanks.

But please clarify. Because you seem to suggest that Resynthesizer depends on PyGTK.  Resynthesizer Python plugins do not "import pygtk" so they don't need it.  The Resynthesizer Python plugins do not implement their own GUI, but rather the GUI dialogs are implemented by PyGimp (when they "import gimpfu"), which uses PyGtk.  The point is, I think that when you installed flatpak GIMP it should have included PyGTK.  It seems strange that you would have separately needed to install flatpak PyGTK.

In other words, the GIMP project is going to a great deal of trouble to flatpak GIMP 2.10, exactly because it uses Gtk2 and PyGTK and Python2, all of which are not supported by most distributions.  So GIMP must include all those in the GIMP flatpak (which is a "platform" with all dependencies.)  That was working, but if it didn't work for you, maybe something has changed.

**Comment by vido89** [comment]
I think that `Python2 support for GIMP` is droped. Please take a look at [this post](https://bbs.archlinux.org/viewtopic.php?id=253710). So I have two versions of GIMP 2.10.24 from flatpak and 2.10.18 from local repo
So when I start 2.10.18 (from repo) it says `/usr/bin/env: ‘python’: No such file or directory` So I have python 2.7 installed and I link it `sudo ln -s /usr/bin/python2.7 /usr/bin/python` and Im getting

```
Traceback (most recent call last):
  File "/usr/lib/gimp/2.0/plug-ins/plugin-uncrop.py", line 46, in <module>
    from gimpfu import *
ImportError: No module named gimpfu****
```

**Comment by vido89** [comment]
Or I just find out that you can make work .deb version

`sudo apt install python2.7 python python-cairo python-gobject-2`

`wget http://archive.ubuntu.com/ubuntu/pool/universe/p/pygtk/python-gtk2_2.24.0-6_amd64.deb`
`wget http://ftp.de.debian.org/debian/pool/main/g/gimp/gimp-python_2.10.8-2_amd64.deb`

`sudo dpkg -i *.deb`
`sudo apt install gimp-plugin-registry`

That should do it :)

**Comment by FalcoGer** [comment]
> Or I just find out that you can make work .deb version
> 
> `sudo apt install python2.7 python python-cairo python-gobject-2`
> 
> `wget http://archive.ubuntu.com/ubuntu/pool/universe/p/pygtk/python-gtk2_2.24.0-6_amd64.deb`
> `wget http://ftp.de.debian.org/debian/pool/main/g/gimp/gimp-python_2.10.8-2_amd64.deb`
> 
> `sudo dpkg -i *.deb`
> `sudo apt install gimp-plugin-registry`
> 
> That should do it :)

That is not a solution, it's a workaround.
Why should I install obsolete software when the proper solution is to update the dependencies of this software? I am glad I finally was able to get rid of python 2. python is a major pain anyway with the millions of libraries and every little bit of python code requires 20 libs with exact versions and they can't coexist unless you fiddle around with virtual environments, which of course beats the whole point of a package manager in the first place.

**Comment by vido89** [comment]
Thats closest that you will get with native .deb way

**Comment by realalexandergeorgiev** [comment]
> I just found that PyGTK hasn't been maintained for 9 years, so it's dropped and breaks Resynthesizer for Gimp 2.10.x. So in order to install it you need to install it from flatpak. After installing GIMP run
> 
> `flatpak install org.gimp.GIMP.Plugin.Resynthesizer`
> 
> but due to the [bug](https://github.com/flatpak/flatpak/issues/4208) you need to copy Resynthesizer from
> 
> ` /var/lib/flatpak/runtime/org.gimp.GIMP.Plugin.Resynthesizer/x86_64/2-3.36/active/files/plug-ins/` to ` /home/your-name/.var/app/org.gimp.GIMP/config/GIMP/2.10/plug-ins/`

this was useful, even on newer ubuntu. i still gave up and it didnt burn. what a chaos and mess this is. restoring my system now :(
